printer/javascript.txt

```diff
@@ -1,30 +0,0 @@
-
-but then
-
-everyone hates C because it's the ultimate footgun. it's physically impossible to write correct code.
-same for C++ except for templating, which is even worse. forgot a semicolon? 111 errors.
-everyone hates javascript because of its random type coercion and poor language design. sort([1, 11, 2]), variable hoisting and objects as maps with prototyping and mixing pre- and self defined properties? also, it's literally impossible to write a program that runs on the first try without errors. wait, i have a better idea ... why not use it on the server? i'll just use npm so i can left-pad.
-everyone hates PHP (which stands for Personal Home Page) because of it's InConsIstent standard_library $namingConventions, security problems (yay for eval($_GET['foo'])) and, oh yes, mixing code and html. it's slow. it's impossible not to write spaghetti code in it. it's not so bad as it once was - the most recent version even introduced partial unicode support!
-everyone hates python for its global interpreter lock and the inability to move on from 2.7. it's like being stuck in the 80s.
-everyone hates ruby because it's too fucking slow and too much dark magic going on. if it wasn't for that abhorrent monstrosity ruby-on-rails, it would have gone the way of the dodo decades ago. ruby used to be the crack epidemic for vulnerable youth developers, but now everybody grew older, wiser, got their life under control, sobered up and thus ditched ruby.
-everyone hates haskell because it's not really usable in the real world (strictly academia only) and either way, nobody really gets what a goddamn monad is. you could specialize in haskell and starve to death in a world where software developers are the most precious resource.
-at least it's not lisp though. (((((help)))))
-everyone hates rust because it's the oh so cool new thing to do and it's so much better, the ultimate language to rule them all and every last piece of software absolutely must be rewritten in rust but, honestly, the borrow checker keeps me from getting anything done. also compile times duh.
-everyone hates D because ... wait, nobody hates D because nobody uses it. also, why settle on D if you could use rust? D is for people who can't escape the sunk cost fallacy.
-everyone hates golang because no generics and you're pretty much at googles mercy if you specialize in it, and we've seen how well that worked out for android devs.
-everyone hates swift and kotlin for not bringing anything new to the table. they maybe improve the worst problems (of java and objective-c) a bit, but in the end, they do nothing other languages haven't done better decades ago. pretty much the only revolutionarily good thing swift did was killing off objective-c. in the land of the blind, the one eyed is kind. java devs love kotlin only because at least it's not java.
```

Comment:
> java devs love kotlin because at least it's not java.
not wrong though

Reply:
D is only good for malware at this point tbh